lib/annotate/annotate_models.rb

```diff
@@ -908,6 +913,58 @@ def get_attributes(column, column_type, klass, options)
 
       attrs
     end
+
+    # Returns the regular expression for finding a column definition based on
+    # the format of the annotations being made
+    #
+    # == Returns:
+    # Regular expression
+    #
+    # @param [Symbol] symbol representation of the format being used for
+    #   annotations
+    def column_pattern_for(format_type)
+      case format_type
+      when :markdown then /^#\s+\*{2}`\w+`\*{2}(?:\s+\|\s`.*`)*$/
+      when :rdoc then
+        field_name_regex = /^#\s+\*\w+\*\:{2}/
+        field_info_regex = /\s+\<tt\>.*\<\/tt\>$/
+
+        /#{field_name_regex}#{field_info_regex}/
+      when :yard
+        attribute_regex = /^#\s@!attribute\s\w+$/
+        return_regex = /^#\s+@return\s+\[\w+\]$/
+
+        /#{attribute_regex}|#{return_regex}/
+      else # :bare/default
+        /^#[\t ]+[\w\*\.\`\@\!\:]+[\t ]+.+$/
```

Comment:
This is the catch-all regular expression. I would like to move it to something more specific to the `bare` format.

My initial implementation broke something for foreign keys, so I will need to see how that is handled for the various formats.